src/main/java/gmailgdogra/service/FormatXlsxService.java

```diff
@@ -0,0 +1,145 @@
+package gmailgdogra.service;
+
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFCellStyle;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.time.LocalTime;
+
+import static gmailgdogra.AppConstants.*;
+
+public class FormatXlsxService {
+
+    private static XSSFWorkbook workbook;
+    private static XSSFSheet sheet;
+
+    public static XSSFWorkbook of(XSSFWorkbook plainXlsx) {
+        workbook = plainXlsx;
+        sheet = workbook.getSheetAt(0);
+        insertEmptyRowAfterEachLocation();
+        applyStyles();
+        return workbook;
+    }
+
+    private static void insertEmptyRowAfterEachLocation() {
+        for (int rowNo = 2; rowNo < sheet.getPhysicalNumberOfRows(); rowNo++) {
+            String previousRowLocation = sheet.getRow(rowNo - 1).getCell(0).getStringCellValue();
+            String currentRowLocation = sheet.getRow(rowNo).getCell(0).getStringCellValue();
+            if (!previousRowLocation.equals(currentRowLocation)) {
+                insertEmptyRowBefore(rowNo);
+                rowNo++;
+            }
+        }
+    }
+
+    private static void insertEmptyRowBefore(int rowNo) {
+        shiftRowsDownByOneFrom(rowNo);
+        Row row = sheet.createRow(rowNo);
+        for (int col = LOCATION_COL_NO; col <= LOGICAL_DEVICE_COL_NO; col++) {
+            row.createCell(col).setBlank();
+        }
+    }
+
+    private static void shiftRowsDownByOneFrom(int rowNo) {
+        sheet.shiftRows(rowNo, sheet.getLastRowNum(), 1);
+    }
+
+    private static void applyStyles() {
+        XSSFCellStyle titleRowStyle = createTitleRowStyle();
+        XSSFCellStyle highlightedRowStyle = createHighlightedRowStyle();
+        XSSFCellStyle normalRowStyle = createNormalRowStyle();
+
+        for (Row row : sheet) {
+
+            int currentRowNo = row.getRowNum();
+
+            if (currentRowNo == TITLE_ROW_NO) {
```

Comment:
You can combine this and the next condition to make the method more compact. Always try to keep your methods under 20 lines of code.